specification/cognitiveservices/data-plane/Language/preview/2022-02-01-preview/textanalytics.json

```diff
@@ -0,0 +1,3090 @@
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AnalyzeTextTaskKind": {
+      "type": "string",
+      "description": "Enumeration of supported Text Analysis tasks.",
+      "enum": [
+        "SentimentAnalysis",
+        "EntityRecognition",
+        "PiiEntityRecognition",
+        "KeyPhraseExtraction",
+        "LanguageDetection",
+        "EntityLinking",
+        "Healthcare",
+        "ExtractiveSummarization",
+        "CustomEntityRecognition",
+        "CustomSingleClassification",
+        "CustomMultiClassification"
+      ],
+      "x-ms-enum": {
+        "name": "AnalyzeTextTaskKind",
+        "modelAsString": true
+      }
+    },
+    "AnalyzeTextTaskResultsKind": {
+      "type": "string",
+      "description": "Enumeration of supported Text Analysis task results.",
+      "enum": [
+        "SentimentAnalysisResults",
+        "EntityRecognitionResults",
+        "PiiEntityRecognitionResults",
+        "KeyPhraseExtractionResults",
+        "LanguageDetectionResults",
+        "EntityLinkingResults"
+      ],
+      "x-ms-enum": {
+        "name": "AnalyzeTextTaskResultsKind",
+        "modelAsString": true
+      }
+    },
+    "AnalyzeTextLROResultsKind": {
+      "type": "string",
+      "description": "Enumeration of supported Text Analysis long-running operation task results.",
+      "enum": [
+        "SentimentAnalysisLROResults",
+        "EntityRecognitionLROResults",
+        "PiiEntityRecognitionLROResults",
+        "KeyPhraseExtractionLROResults",
+        "LanguageDetectionLROResults",
+        "EntityLinkingLROResults",
+        "HealthcareLROResults",
+        "ExtractiveSummarizationLROResults",
+        "CustomEntityRecognitionLROResults",
+        "CustomSingleClassificationLROResults",
+        "CustomMultiClassificationLROResults"
+      ],
+      "x-ms-enum": {
+        "name": "AnalyzeTextLROResultsKind",
+        "modelAsString": true
+      }
+    },
+    "MultiLanguageAnalysisInput": {
+      "properties": {
+        "documents": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/MultiLanguageInput"
+          }
+        }
+      }
+    },
+    "LanguageDetectionAnalysisInput": {
+      "properties": {
+        "documents": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/LanguageInput"
+          }
+        }
+      }
+    },
+    "AnalyzeTextTask": {
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeTextTaskKind"
+        }
+      }
+    },
+    "AnalyzeTextTaskResult": {
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeTextTaskResultsKind"
+        }
+      }
+    },
+    "AnalyzeTextEntityLinkingInput": {
```

Comment:
Is this type intended for sync input? Is the EntityLinkingTask type intended for async input?

Reply:
Yes, that's correct.  I've just made a few changes to hopefully make it more clear which task types are supported in which scenario (sync vs async).  I basically just added a new enum type for LRO tasks and made sure to have the LRO task input types inherit from an object which uses this new enum as the discriminator value.  Let me know if this helps.